Common/Symbol.cs

```diff
@@ -84,15 +87,85 @@ public static Symbol Create(string ticker, SecurityType securityType, string mar
+        /// <param name="expiry">The option expiry date</param>
+        /// <param name="alias">An alias to be used for the symbol cache. Required when 
+        /// adding the same security from diferent markets</param>
+        /// <returns>A new Symbol object for the specified option contract</returns>
+        public static Symbol CreateOption(Symbol underlyingSymbol, string market, OptionStyle style, OptionRight right, decimal strike, DateTime expiry, string alias = null)
+        {
+            var sid = SecurityIdentifier.GenerateOption(expiry, underlyingSymbol.ID, market, strike, right, style);
+    
+            if (expiry == SecurityIdentifier.DefaultDate)
+            {
+                alias = alias ?? "?" + underlyingSymbol.Value.ToUpper();
+            }
+            else
+            {
+                var sym = underlyingSymbol.Value;
+                if (sym.Length > 5) sym += " ";
+
+                alias = alias ?? SymbolRepresentation.GenerateOptionTickerOSI(sym, sid.OptionRight, sid.StrikePrice, sid.Date);
+            }
+
+            return new Symbol(sid, alias, underlyingSymbol);
+        }
+
+        /// <summary>
+        /// Provides a convenience method for creating a futures Symbol.
+        /// </summary>
+        /// <param name="underlying">The underlying ticker</param>
+        /// <param name="market">The market the underlying resides in</param>
+        /// <param name="expiry">The option expiry date</param>
+        /// <param name="alias">An alias to be used for the symbol cache. Required when 
+        /// adding the same security from different markets</param>
+        /// <returns>A new Symbol object for the specified option contract</returns>
+        public static Symbol CreateFuture(string underlying, string market, DateTime expiry, string alias = null)
+        {
+            var underlyingSid = SecurityIdentifier.GenerateBase(underlying, market);
+            var sid = SecurityIdentifier.GenerateFuture(expiry, underlyingSid, market);
+            var underlyingSymbol = new Symbol(underlyingSid, underlying);
+
+            if (expiry == SecurityIdentifier.DefaultDate)
+            {
+                alias = alias ?? "?" + underlying.ToUpper();
+            }
+            else
+            {
+                var sym = sid.Symbol;
+                alias = alias ?? SymbolRepresentation.GenerateFutureTicker(sym, sid.Date);
+            }
+
+            return new Symbol(sid, alias, underlyingSymbol);
+        }
+
+
+        /// <summary>
+        /// Method returns true, if symbol is a derivative canonical symbol
+        /// </summary>
+        /// <returns>true, if symbol is a derivative canonical symbol</returns>
+        public bool IsCanonical()
+        {
+            return (ID.SecurityType == SecurityType.Future || ID.SecurityType == SecurityType.Option) &&
+                   HasUnderlying && 
+                   ID.Date == SecurityIdentifier.DefaultDate;
```

Comment:
Perhaps it makes sense for `SecurityIdentifier` to define what it means to be canonical, since `Symbol` is user facing wrapper around `SecurityIdentifier`, I would expect this method to just pass through. Also, curiou why we chose a method here instead of a property?